src/site/antora/modules/ROOT/pages/manual/customloglevels.adoc

```diff
@@ -47,18 +64,106 @@ Predefined levels are available for Log4j API integrators through link:../javado
+link:../javadoc/log4j-api/org/apache/logging/log4j/Logger.html#atTrace()[Logger.atTrace()]
+
+|===
+
+By using shorthand methods, you can rewrite the example above as:
+
+[source,java,indent=0]
+----
+include::example$manual/customlevels/LevelExample.java[tag=shorthand]
+----
+
+[#level-selection]
+=== Which level to use?
+
+[CAUTION]
+====
+While Log4j API defines a set of standard levels, it does not define the purpose of these levels.
+Many different conventions on which log levels to use coexist in the industry.
+When in doubt, you should ask your teammates about the convention used at your company.
+====
+
+Most log level usage conventions divide log levels into two categories:
+
+* the most severe log levels (e.g. `FATAL`, `ERROR` and `WARN`) are used to inform the system administrator about a problem in the Java application that needs to be fixed.
+The more severe the problem, the more severe the log level.
++
+Log events with these levels should be used sparingly and should allow the system administrator to fix the problem.
+
+* the less severe log levels (e.g. `INFO`, `DEBUG`, `TRACE`) provide context that allow a system administrator or developer to diagnose the reason of an application failure.
+The most severe of them describe events that concern the whole application, while the less severe describe events that are interesting for a single sub-system.
+
+[#DefiningLevelsInCode]
+=== Custom log levels
+
+While most Java logging APIs adopt the same set of standard logging levels, some logging APIs, such as xref:log4j-jul.adoc#default-level-conversions[JUL]
+and external logging systems, such as
+https://datatracker.ietf.org/doc/html/rfc5424#section-6.2.1[Syslog]
+and
+https://opentelemetry.io/docs/specs/otel/logs/data-model/#displaying-severity[OpenTelemetry]
+support additional logging levels that can not be mapped to the standard ones.
 
-[source,java]
+To improve interoperability between logging systems, Log4j API supports custom log levels that can be defined using the
+link:../javadoc/log4j-api/org/apache/logging/log4j/Level.html#forName(java.lang.String,int)[`Level.forName()`]
+method:
+
+[source,java,indent=0]
+----
+include::example$manual/customlevels/LevelExample.java[tag=custom-definition]
+----
+
+Custom log levels can be used in your code with the usual `Logger.log(..)` and `Logger.atLevel(Level)` methods:
+
+[source,java,indent=0]
 ----
-LOGGER.info("Hello, {}!", userName); // <1>
-LOGGER.log(Level.INFO, "Hello, {}!", userName); // <2>
+include::example$manual/customlevels/LevelExample.java[tag=custom]
 ----
-<1> Using `Logger#info()` to log a message at `INFO` level
-<2> Using `Logger#log()` to log a message and specifying the `INFO` level explicitly
 
 There are several ways levels can be employed in the Log4j configuration file for filtering purposes.
 Filtering on levels for loggers is a pretty common example:
+
 [source,xml]
```

Comment:
Shall we externalize this configuration share it in all 4 formats?

Reply:
Nice catch, I meant to get rid of this Log4j Core specific example, since there is one in the section below.

Fixed in 4f2434bcd6934d17267cd9e5187ac9e335cec096.